### PR TITLE
test: add tests for asm instructions without whitespace

### LINKED ISSUE #1268: Add grammar tests for whitespace after `asm` instructions before `}`
The new parser merged in #1258 fixed issue #1265, but grammar tests are missing.

### PR DESCRIPTION
**Pull request by xpyctumo**
## Issue

Closes #1268

## Checklist

- [ ] I have updated CHANGELOG.md
- [ ] I have run all the tests locally and no test failure was reported
- [x] I have run the linter, formatter and spellchecker
- [x] I did not do unrelated and/or undiscussed refactorings

~~**Also, we need fix old parser to handle this tests**~~
UPD: As a solution - tests that only require to use **new parser** should be placed in `src/grammar/test/next/test`

### CODE REVIEW

File: src/grammar/next/test-failed/asm-instruction-invalid-syntax.tact [hunk 1]
```diff
@@ -0,0 +1,3 @@
+asm fun Bar(): Int {-ROT ADD MUL};
```
Comment (anton-trunov) [inline src/grammar/next/test-failed/asm-instruction-invalid-syntax.tact]:
```suggestion
asm fun Bar(): Int {-ROT ADD MUL}
```

File: src/grammar/next/grammar.spec.ts [hunk 2]
```diff
@@ -9,21 +9,32 @@ expect.addSnapshotSerializer({
...
-            const { parse } = getParser(ast, "new");
-            expect(parse(r.code, "<unknown>", "user")).toMatchSnapshot();
-        });
+    const shouldParsePaths = [__dirname + "/../test/", __dirname + "/test/"];
+
+    for (const path of shouldParsePaths) {
+        for (const r of loadCases(path)) {
+            it("should parse " + r.name, () => {
+                const ast = getAstFactory();
+                const { parse } = getParser(ast, "new");
```
Comment (anton-trunov) [inline src/grammar/next/grammar.spec.ts:18]: this should be moved out of the loop and also reused in the new tests for the new parser

Comment (anton-trunov) [reply src/grammar/next/grammar.spec.ts:18]: ah, this suggestion is incorrect, since `id`s would be shared

File: src/grammar/next/test-failed/asm-instruction-invalid-syntax.tact (hunk 1 above)
Comment (xpyctumo) [reply src/grammar/next/test-failed/asm-instruction-invalid-syntax.tact]: Thats an example of failed test.  Just so there's a `test-failed` folder.

Comment (xpyctumo) [reply src/grammar/next/test-failed/asm-instruction-invalid-syntax.tact]: Do you think we don't need to do `test-failed` separately?

**Comment by xpyctumo** [comment]
Ouch. CI on Windows was failed. Quite a surprise

**Comment by reverofevil** [comment]
> Ouch. CI on Windows was failed. Quite a surprise

At the moment it's expected

File: src/grammar/next/test-failed/asm-instruction-invalid-syntax.tact (hunk 1 above)
Comment (reverofevil) [reply src/grammar/next/test-failed/asm-instruction-invalid-syntax.tact:1]: Probably we should do a different kind of failing test, because in this case it's not the asm instruction parser that is failing.

Maybe something like `asm fun Bar() Int`.

Comment (reverofevil) [reply src/grammar/next/test-failed/asm-instruction-invalid-syntax.tact:1]: Oh wait, we already have a `test-failed` folder, this test is not related to whitespaces, and I don't see why would we need this test.

Comment (verytactical) [reply src/grammar/next/test-failed/asm-instruction-invalid-syntax.tact]: We already have `test-failed` folder, and this test is not about whitespace.

Comment (xpyctumo) [reply src/grammar/next/test-failed/asm-instruction-invalid-syntax.tact]: I removed that

**Review by verytactical — APPROVED** [review]
